doc/md/writing-motoko/candid.md

```diff
@@ -0,0 +1,135 @@
+These tools further enhance the developer experience and promote interoperability.
+
+Importantly, Candid's design allows for backwards-compatible upgrades of canister interfaces.
+This feature facilitates the evolution of services over time, a critical aspect for long-lived applications
+on the Internet Computer.
+
+In essence, Candid forms a fundamental part of the IC ecosystem. It provides a standardized way to describe, serialize, and communicate data between different parts of the platform. Its seamless integration with Motoko and other languages makes it easier for developers to create interoperable and type-safe services on the Internet Computer.
+
+## Explicit Candid serialization
+
+Motoko's `to_candid` and `from_candid` operators allow you to work with Candid-encoded data.
+
+`to_candid (<exp1>, ..., <expn>)` serializes a sequence of Motoko values into a `Blob` containing a Candid binary encoding of the data.
+
+For example,
+
+``` motoko no-repl
+let encoding : Blob = to_candid ("dogs", #are, ['g', 'r', 'e', 'a', 't']);
+```
+
+`from_candid <exp>` deserializes a blob that contains Candid data back into a Motoko value.
+
+``` motoko no-repl
+ let ?(t, v, cs) = from_candid encoding : ?(Text, {#are; #are_not}, [Char]);
+```
+
+`from_candid` will trap if its arguments is a blob that does not contain valid Candid data.
+Because deserialization can fail if the encoded value does not have the expected Candid type,
+`from_candid` returns a value of option type, with `null` indicated the encoding is well-formed but of the wrong Candid type or some value `?v`
+where `v` is the decoded value.
+`from_candid` can only be used in the context of other code that determines its optinal result type, for which a type annotation may be required.
+
+For example, this code that underspecifies the expected type of the decoded value, is rejected by the compiler:
+
+``` motoko no-repl
+let ?(t, v, cs) = from_candid encoding;
+```
+
+The `to_candid` and `from_candid` operators are keywords, built into the language and handle most common use cases automatically.
+The operators ensure type safety and proper data encoding without requiring developers to manually handle the intricacies of Candid serialization.
+
+:::danger
+
+Although `to_candid` will return a valid Candid encoding of its argument, there are actually many different Candid encodings, and thus blobs, for the same value.
+There is no guarantee that `to_candid` will always return the same `blob`, given the same argument.
+That means that you should never use these blobs to compare values for equality, or be tempted to
+compute a hash for a value, by hashing its Candid encoding.
+The hash of a value should be unique, but if you compute it from one of several Candid encodings, it may not be.
+
+:::
+
+See the language manual for more details on [`Candid serialization`](../reference/language-manual#candid_serialization).
```

Comment:
```suggestion
The `to_candid` and `from_candid` operators are keywords built into the language and handle most common use cases automatically.
The operators ensure type safety and proper data encoding without requiring developers to manually handle the intricacies of Candid serialization.

:::danger

Although `to_candid` will return a valid Candid encoding of its argument, there are actually many different Candid encodings, and thus blobs, for the same value.
There is no guarantee that `to_candid` will always return the same `blob`, given the same argument.
That means that you should never use these blobs to compare values for equality or be tempted to
compute a hash for a value by hashing its Candid encoding.
The hash of a value should be unique, but if you compute it from one of several Candid encodings, it may not be.

:::

See the language manual for more details on [`Candid serialization`](../reference/language-manual#candid_serialization).
```